### ISSUE TITLE
Comboboxes with "Flat" and "Popup" flat styles are not displayed in High Contrast mode

### ISSUE THREAD

**Issue by SergeySmirnov-Akvelon** [dotnet/winforms#6155]
.NET Core Version: 
- 7.0.0-alpha.1.21556.5

Have you experienced this same bug with .NET Framework?: 
- Yes

**Problem description:**
1. Create a `Form` with a `Combobox`
2. Set `Popup` or `Flat` flat style
3. Turn on High Contrast mode
4. Run the application

**Actual behavior:**
Comboboxes should be displayed are not displayed in High Contrast mode:
![image](https://user-images.githubusercontent.com/23376742/141124465-cbfe182c-2edc-42a4-a679-0bfcb2041c8b.png)

**Expected behavior:**
Comboboxes should be displayed in High Contrast mode.

**Comment by jbhensley** [comment]
It looks like the `ComboBox` border gets painted using `SystemColors.Window` when in `Flat` or `Popup` mode. But since the inside of the control is the same, there's no outline. This seems to raise a larger question of whether the `ComboBox` should actually have a "flat" mode in high contrast. It would be easy enough to add `&& !DisplayInformation.HighContrast` to the following check, which should result in it being rendered the same as `FlatStyle.Standard` in that scenario.

https://github.com/dotnet/winforms/blob/682a47e3414abfa65a742b2db444f468342e9e4b/src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.cs#L3905-L3907

**Comment by MelonWang1** [comment]
Verified in the .NET 7.0 SDK build: 7.0.100-preview.4.22227.3, it is fixed.
![61554](https://user-images.githubusercontent.com/94418985/165691466-f0316fff-fca3-42fa-b420-b4c638ebffe0.png)
